### ISSUE TITLE
General error: 1215 Cannot add foreign key constraint

### ISSUE THREAD

**Issue by erictracy11492** [susanBuck/dwa15-fall2017#229]
When attempting to rename my pivot table, I am receiving this error:
<img width="562" alt="screen shot 2017-12-11 at 7 14 23 pm" src="https://user-images.githubusercontent.com/31494476/33860580-8a50ed9a-dea7-11e7-9710-00afd4bae5b5.png">

<img width="527" alt="screen shot 2017-12-11 at 7 14 55 pm" src="https://user-images.githubusercontent.com/31494476/33860595-9c5dc634-dea7-11e7-8d69-4a2d1d7f68e8.png">
<img width="633" alt="screen shot 2017-12-11 at 7 15 17 pm" src="https://user-images.githubusercontent.com/31494476/33860613-a9f0380e-dea7-11e7-8063-15d9991d4832.png">
<img width="453" alt="screen shot 2017-12-11 at 7 15 53 pm" src="https://user-images.githubusercontent.com/31494476/33860630-bfc877e0-dea7-11e7-9403-046ac2d04f2a.png">

**Comment by ysrinu** [comment]
@erictracy11492 
Does that mean, everytime the migration script is run, it will first create task_user pivot table and then rename to tag_user_task?
Can you modify migration so it will only create the tag_user_task and not tag_user?

Also, in your models that are related to the pivot table, in the  belongsToMany relationship, you may have to define foreign key and local key names since eloquent may not be able to figure them anymore due to non-standard naming conventions (I suspect).

Can you also check if you have duplicate foreign key indexes? 
My pivot table migration script creates foreign indexes and then references foreign keys

(I am using mediumInteger unsigned for my use case)

```
public function up()
{
    Schema::create('asset_keyword', function (Blueprint $table) {
        $table->increments('id');
        $table->timestamps();
        $table->mediumInteger('asset_id')->unsigned();
        $table->mediumInteger('keyword_id')->unsigned();

        $table->index(["asset_id"], 'fk_asset_id_idx3');
        $table->index(["keyword_id"], 'fk_keyword_id_idx');

        $table->foreign('asset_id', 'fk_asset_id_idx3')
        ->references('id')->on('assets')
        ->onDelete('no action')
        ->onUpdate('no action');
        $table->foreign('keyword_id', 'fk_keyword_id_idx')
        ->references('id')->on('keywords')
        ->onDelete('no action')
        ->onUpdate('no action');
    });
}
```

**Comment by erictracy11492** [comment]
@ysrinu 
I modified the initial migration. I was able to remove of the duplicate table, but when trying to migrate I am now getting this error:
<img width="565" alt="screen shot 2017-12-11 at 8 43 25 pm" src="https://user-images.githubusercontent.com/31494476/33863086-142698a6-deb4-11e7-8b30-73f24d230fe7.png">

**Comment by ysrinu** [comment]
@erictracy11492  try `php artisan migrate:fresh`, so it will first drop all tables and then recreate

**Comment by erictracy11492** [comment]
@ysrinu
And when I attempt to seed the tables I am getting this error:
<img width="563" alt="screen shot 2017-12-11 at 9 04 29 pm" src="https://user-images.githubusercontent.com/31494476/33863687-fb2c784a-deb6-11e7-8096-aeb751688d3f.png">
I do not have a column called 'user_tasks_id' it is instead called 'task_id'. Do I need to change any of this below?
<img width="553" alt="screen shot 2017-12-11 at 9 06 55 pm" src="https://user-images.githubusercontent.com/31494476/33863734-4086311a-deb7-11e7-9972-152910f6b260.png">

**Comment by ysrinu** [comment]
@erictracy11492 You may still have the old column name used in your seeder insert statement

**Comment by ysrinu** [comment]
@erictracy11492 is this fixed?

**Comment by erictracy11492** [comment]
@ysrinu 
Not quite. I manually changed the name of the migration. It let me migrate using php artisan:migrate fresh. When attempting to seed, I received this error again:
<img width="561" alt="screen shot 2017-12-12 at 6 32 18 pm" src="https://user-images.githubusercontent.com/31494476/33914314-9b88b2ea-df6b-11e7-9367-a4822bfbafb3.png">
I realized the user_task table it references has a plural model name:
<img width="606" alt="screen shot 2017-12-12 at 6 38 13 pm" src="https://user-images.githubusercontent.com/31494476/33914334-b9bda4b4-df6b-11e7-9518-c75e2fce603c.png">
Could this be causing it?
